src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/UseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -85,16 +85,11 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context, ImmutableHashSe
 
             ITypeSymbol paramType = symbol.Parameters[0].Type;
 
-            if (paramType.TypeKind == TypeKind.TypeParameter)
+            if (paramType.TypeKind is TypeKind.TypeParameter or TypeKind.Enum)
             {
                 return;
             }
 
-            if (paramType.TypeKind == TypeKind.Enum)
-            {
-                paramType = ((INamedTypeSymbol)paramType).EnumUnderlyingType;
```

Comment:
Current implementation matches the old FxCop implementation (internal source):

```csharp
    public override ProblemCollection Check(Member member)
    {
      PropertyNode property = member as PropertyNode;
      if (property == null) return null; 
      
      if (!RuleUtilities.IsIndexer(property))
        return null;
      
      Method getMethod = property.Getter;
      if (getMethod == null || getMethod.Parameters.Count != 1)
        return null;
        
      TypeNode indexerType = getMethod.Parameters[0].Type;
        EnumNode enumNode = indexerType as EnumNode;

      if (enumNode != null) 
        indexerType = enumNode.UnderlyingType;

      if (indexerType != FrameworkTypes.Int32 
        && indexerType != FrameworkTypes.Int64  
        && indexerType != FrameworkTypes.Object 
        && indexerType != FrameworkTypes.String
        && !(indexerType is TypeParameter || //Generic indexers (e.g. this[T t]) are ok
        indexerType is ClassParameter)
        && !RuleUtilities.IsOverride(getMethod))
      {
        Resolution resolution = GetResolution( 
                            // Use an integral value or string
          property          // as the parameter to indexer {0}.
          );
        Problem problem = new Problem(resolution);
        Problems.Add(problem);
      }

      return Problems;
    }
```

Reply:
Is it a must to match FxCop exactly (especially when the change produces less warnings rather than new warnings)?

I don't know if the Framework Design Guidelines were updated at some point to allow `enum`s when it didn't allow it in past. So, probably the legacy implementation was written with respect to old version of Framework Design Guidelines?

Reply:
I don't have quick access to a digital copy of the first edition of FDG, but I can say that "AVOID indexers with parameter types other than System.Int32, System.Int64, System.String, System.Object, or an enum" is the wording from second edition (2009).  We haven't yet been able to update the site for 3rd edition content for reasons (that to lawyers are significant, and to laymen are insignificant... but... still).

In third edition (2020) it became "AVOID indexers with parameter types other than System.Int32, System.Int64, System.String, System.Range, or an enum; except on dictionary-like types."  (Object moved to "except on dicitonary-like types" and Range got added)